python/pyspark/sql/dataframe.py

```diff
@@ -5984,6 +5984,41 @@ def writeTo(self, table: str) -> DataFrameWriterV2:
         """
         ...
 
+    @dispatch_df_method
+    def mergeInto(self, table: str, condition: "ColumnOrName") -> MergeIntoWriter:
+        """
+        Merges a set of updates, insertions, and deletions based on a source table into
+        a target table.
+
+        .. versionadded:: 4.0.0
+
+        Parameters
+        ----------
+        table : str
+            Target table name to merge into.
+        condition : :class:`Column`
```

Comment:
This is `Column`, but the signature is `ColumnOrName`?

Reply:
This actually must be a `Column` under the hood -- when given a string it must be wrapped by an `expr()`. What's your suggestion here? Should we allow strings (and wrap them) to simplify the user experience, or ask the user to explicitly convert them to Columns?

Reply:
I changed the method to accept `Column`.

Reply:
Thank you.